File: LeetCode/27.Remove-Element/Mario/27. Remove Element.py
#!/usr/bin/env python3

# 27. Remove Element
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i, j = 0, len(nums)-1
        while i <= j:
            if nums[i] != val:
                i += 1
            else:
                nums[i], nums[j] = nums[j], nums[i]
                j -= 1
        return i

File: LeetCode/27.Remove_Element/27.Remove_Element.swift
class Solution {
    func removeElement(_ nums: inout [Int], _ val: Int) -> Int {
        var write = 0

        for (i, num) in nums.enumerated() {
            if num != val {
                nums[write] = num
                write += 1
            }
        }
        return write
    }
}
